### PR TITLE
SMG Gun+Mag Holster

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

Adds a new holster for the SMG. It can hold a M39 SMG plus two M39 magazines, it replaces the old SMG belt. 

# Explain why it's good for the game

The M39 SMG is intended as an inferior firearm to the mainline M41A. The perceived benefit of a M39 would be its lightness compared to the bulkier M41A, however this is not reflected in-game outside of the M39 not confering a movement speed down when wielded, which is shared with many other bulky guns like the M4RA.

The idea around this new holster is that it would give Marines a better reason to consider taking the M39, as Marines in a more support oriented role can access a unique holster that lets them carry additional magazines in addition to the firearm itself. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Screenshot 2023-07-07 19 17 20
Current iteration

Screenshot 2023-07-07 01 58 02
Screenshot 2023-07-07 02 01 46
Old iteration

Screenshot 2023-07-06 17 17 56
Old sprite and internal storage view

OBJ sprites and the inventory capacity as seen in game

</details>

# Changelog
:cl:
add: The SMG belt has been changed. It now holds a SMG plus two extra magazines, plus it can fit in either the belt or armour slot. 
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/marine_armor.dm [hunk 1]
```diff
@@ -579,6 +579,7 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	armor_rad = CLOTHING_ARMOR_NONE
 	armor_internaldamage = CLOTHING_ARMOR_NONE
 	storage_slots = 3
+	slowdown = SLOWDOWN_ARMOR_VERY_LIGHT
```
Comment (Zonespace27) [inline code/modules/clothing/suits/marine_armor.dm:582]: You have a polluted master

Comment (Steelpoint) [reply code/modules/clothing/suits/marine_armor.dm:582]: oh god

**Review by sleepynecrons — CHANGES_REQUESTED** [review]
use the m39 holster that we already have as a base. sticking the m39 in a pistol holster just looks extremely silly and is visually inconsistent with the other holsters.
image

**Comment by Steelpoint** [comment]
> use the m39 holster that we already have as a base. sticking the m39 in a pistol holster just looks extremely silly and is visually inconsistent with the other holsters. image

I considered that at first, that's even how I initially did it, but the way the code works for the sprites is that the gun is placed behind the sprite not in front of it. So I settled on what is currently being used. 

Outside of a code solution to make the gun sprite appear in front of the rig sprite, which I won't be able to do by myself, I did a hacky fix where I edited the rig sprite to remove the parts of the rig that would be in front of the gun sprite. Not a great solution though.

**Comment by Steelpoint** [comment]
For the interim I've implemented an alt solution as described above, here is how the sprite looks in game

Screenshot 2023-07-07 01 58 02

**Comment by sleepynecrons** [comment]
> For the interim I've implemented an alt solution as described above, here is how the sprite looks in game
> 
> Screenshot 2023-07-07 01 58 02 Screenshot 2023-07-07 02 01 46

swap the holster and mag pouches position and it'll be good to go
image
basically like this

**Comment by Steelpoint** [comment]
Swapped the sprite around
Screenshot 2023-07-07 19 17 20

File: code/game/machinery/vending/vendor_types/intelligence_officer.dm [hunk 2]
```diff
@@ -78,6 +78,9 @@ GLOBAL_LIST_INIT(cm_vending_clothing_intelligence_officer, list(
 		list("M276 Shotgun Shell Loading Rig", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 		list("M276 Toolbelt Rig (Full)", 0, /obj/item/storage/belt/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
 
+		list("ARMOR SLOT (CHOOSE 1)", 0, null, null, null),
+		list("M276 SMG Armor Rig", 0, /obj/item/storage/belt/gun/m39, MARINE_CAN_BUY_ARMOR_SLOT, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/intelligence_officer.dm:82]: Make buyable

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm [hunk 3]
```diff
@@ -125,6 +125,9 @@ GLOBAL_LIST_INIT(cm_vending_clothing_engi, list(
 		list("M276 Toolbelt Rig (Full)", 0, /obj/item/storage/belt/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_MANDATORY),
 		list("M276 M40 Grenade Rig", 0, /obj/item/storage/belt/grenade, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 
+		list("ARMOR SLOT (CHOOSE 1)", 0, null, null, null),
+		list("M276 SMG Armor Rig", 0, /obj/item/storage/belt/gun/m39, MARINE_CAN_BUY_ARMOR_SLOT, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm:129]: Make buyable

File: code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm [hunk 4]
```diff
@@ -139,6 +139,9 @@ GLOBAL_LIST_INIT(cm_vending_clothing_medic, list(
 		list("M276 Shotgun Shell Loading Rig", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 		list("M276 M40 Grenade Rig", 0, /obj/item/storage/belt/grenade, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 
+		list("ARMOR SLOT (CHOOSE 1)", 0, null, null, null),
+		list("M276 SMG Armor Rig", 0, /obj/item/storage/belt/gun/m39, MARINE_CAN_BUY_ARMOR_SLOT, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm:143]: make buyable

File: code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm [hunk 5]
```diff
@@ -96,6 +96,9 @@ GLOBAL_LIST_INIT(cm_vending_clothing_specialist, list(
 		list("M276 M82F Holster Rig", 0, /obj/item/storage/belt/gun/flaregun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 		list("M276 Shotgun Shell Loading Rig", 0, /obj/item/storage/belt/shotgun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 
+		list("ARMOR SLOT (CHOOSE 1)", 0, null, null, null),
+		list("M276 SMG Armor Rig", 0, /obj/item/storage/belt/gun/m39, MARINE_CAN_BUY_ARMOR_SLOT, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm:100]: make buyable

File: code/game/objects/items/storage/belt.dm [hunk 6]
```diff
@@ -1122,6 +1122,23 @@
 	for(var/i = 1 to storage_slots - 1)
 		new /obj/item/ammo_magazine/pistol/highpower/black(src)
 
+/obj/item/storage/belt/gun/m39
+	name = "\improper M276 pattern M39 armor rig"
+	desc = "Special issue variant of the M276 designed to holster a M39 submachine gun and two spare magazines. Uncommonly issued to USCM support and specialist personnel, can only be clipped to a set of armor."
+	icon_state = "m39_armor"
+	item_state = "s_marinebelt"
+	flags_equip_slot = SLOT_SUIT_STORE
```
Comment (morrowwolf) [inline code/game/objects/items/storage/belt.dm:1130]: Don't do this here, add this to the base marine armor in marine_armor.dm in "allowed" line 100

File: code/__DEFINES/vendors.dm [hunk 7]
```diff
@@ -7,6 +7,7 @@
 #define MARINE_CAN_BUY_BACKPACK "backpack"
 #define MARINE_CAN_BUY_POUCH "pouch"
 #define MARINE_CAN_BUY_BELT "belt"
+#define MARINE_CAN_BUY_ARMOR_SLOT "armor_slot"
```
Comment (morrowwolf) [inline code/__DEFINES/vendors.dm:10]: Remove since unneeded after requested changes

File: code/__DEFINES/vendors.dm [hunk 8]
```diff
@@ -20,7 +21,7 @@
...
+#define MARINE_CAN_BUY_ALL list(MARINE_CAN_BUY_UNIFORM = 1, MARINE_CAN_BUY_SHOES = 1, MARINE_CAN_BUY_HELMET = 1, MARINE_CAN_BUY_ARMOR = 1, MARINE_CAN_BUY_GLOVES = 1, MARINE_CAN_BUY_EAR = 1, MARINE_CAN_BUY_BACKPACK = 1, MARINE_CAN_BUY_POUCH = 2, MARINE_CAN_BUY_BELT = 1, MARINE_CAN_BUY_ARMOR_SLOT = 1, MARINE_CAN_BUY_GLASSES = 1, MARINE_CAN_BUY_MASK = 1, MARINE_CAN_BUY_ESSENTIALS = 1, MARINE_CAN_BUY_SECONDARY = 1, MARINE_CAN_BUY_ATTACHMENT = 1, MARINE_CAN_BUY_MRE = 1, MARINE_CAN_BUY_ACCESSORY = 1, MARINE_CAN_BUY_COMBAT_SHOES = 1, MARINE_CAN_BUY_COMBAT_HELMET = 1, MARINE_CAN_BUY_COMBAT_ARMOR = 1, MARINE_CAN_BUY_KIT = 1, MARINE_CAN_BUY_DRESS = 99)
```
Comment (morrowwolf) [inline code/__DEFINES/vendors.dm:24]: Revert since unneeded after requested changes

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
Requested changes made.

**Review by harryob — APPROVED** [review]
code

**Review by nauticall — APPROVED** [review]
sprites 👍

**Comment by Steelpoint** [comment]
gonna make a change

**Comment by Steelpoint** [comment]
Stated to make this replace the current SMG belt, will need to make some changes, shouldn't take too long (famous last words)

**Comment by Steelpoint** [comment]
This SMG belt now replaces all regular instances of the old SMG belt. Old SMG belt is still in the game, though I believe only PMC's use it now.

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**